Support multi-node notebook launching (#1913)
huggingface/accelerate: docs/source/basic_tutorials/notebook.md

notebook_launcher(training_loop, args, num_processes=2)
```

In the case of running on multiple nodes, you need to set up a Jupyter session at each node and run the launching cell at the same time.

For an environment containing 2 nodes (computers) with 4 GPUs each and the main computer with an IP address of "172.31.43.8", it would look like so:

```python
notebook_launcher(training_loop, args, master_addr="172.31.43.8", node_rank=0, num_nodes=2, num_processes=8)

> 

Reviewer: ````suggestion
```

````
Need closing ticks

And in the second Jupyter session on the other machine:

> <Tip>
>     Notice how the `node_rank` has changed
> </Tip>

Reviewer: ```suggestion
<Tip>

    Notice how the `node_rank` has changed
    
</Tip>
```

Small nit here, our docs require a full space between it

```python
notebook_launcher(training_loop, args, master_addr="172.31.43.8", node_rank=1, num_nodes=2, num_processes=8)

> 

Reviewer: ````suggestion
```

````

Here too

notebook_launcher(training_loop, args, num_processes=2)
```

> In the case of running on multiple nodes, you need to set up a jupyter session at each node and run the launching cell at the same time,
> or use [papermill](https://papermill.readthedocs.io/en/latest/) to execute the notebook.

Reviewer: ```suggestion
In the case of running on multiple nodes, you need to set up a Jupyter session at each node and run the launching cell at the same time.
```

> For an environment containing 2 nodes with 4 GPUs per node, the master launcher of the first node with address "172.31.43.8" would look like so:

Reviewer: ```suggestion
For an environment containing 2 nodes (computers) with 4 GPUs each and the main computer with an IP address of "172.31.43.8", it would look like so:
```

> ```python
> 
> notebook_launcher(training_loop, args, master_addr="172.31.43.8", node_rank=0, num_nodes=2, num_processes=8)
> 
> ```

Reviewer: ```suggestion
```python
notebook_launcher(training_loop, args, master_addr="172.31.43.8", node_rank=0, num_nodes=2, num_processes=8)
```
```

> And the second launcher of the node would like so:

Reviewer: ```suggestion
And in the second Jupyter session on the other machine: 

<Tip>
    Notice how the `node_rank` has changed
</Tip>
```

> ```python
> 
> notebook_launcher(training_loop, args, master_addr="172.31.43.8", node_rank=1, num_nodes=2, num_processes=8)
> 
> ```

Reviewer: ```suggestion
```python
notebook_launcher(training_loop, args, master_addr="172.31.43.8", node_rank=1, num_nodes=2, num_processes=8)
```
```
